pipeline/merger/src/main/scala/uk/ac/wellcome/platform/merger/rules/FieldMergeRule.scala

```diff
@@ -0,0 +1,77 @@
+package uk.ac.wellcome.platform.merger.rules
+import uk.ac.wellcome.models.work.internal.{
+  TransformedBaseWork,
+  UnidentifiedWork
+}
+import uk.ac.wellcome.platform.merger.rules.WorkFilters.WorkFilter
+
+case class MergeResult[T](field: T, redirects: Seq[TransformedBaseWork])
+
+/*
+ * A trait to extend in order to merge fields of the type member `Field`
+ *
+ * The implementor must provide the method `merge`, which takes a target
+ * work and a list of source works, and returns the new value of the field
+ * as well as a list of source works that this rule would like to be redirected.
+ *
+ * Many merge rules will want to apply different logic given conditions on
+ * both the target and the source works.
+ * The PartialRule trait is provided to achieve this: in addition to
+ * a `rule` that has the same signature as the `merge` method, the implementor
+ * must provide predicates for valid targets and sources.
+ *
+ * Because PartialRule is a PartialFunction, the `rule` will never be called
+ * for targets/sources that don't satisfy these predicates, and we gain things
+ * like `orElse` and `andThen` for free.
+ */
+trait FieldMergeRule {
+  protected final type Params = (UnidentifiedWork, Seq[TransformedBaseWork])
+  protected type Field
```

Comment:
At the risk of being pedantic: when I first saw this type I assumed that we would somehow encode the name and value into the type, rather than just the type of data. Maybe `FieldData` would be a more accurate description, as a `Field` to me indicates a mapping between two things? But maybe I'm over thinking things..

Reply:
Yeah, agreed, I initially called it `FieldType` but putting "type" in the name of a type felt bad - `FieldData` is good